sentry-core/src/main/java/io/sentry/core/transport/HttpTransport.java

```diff
@@ -101,13 +104,33 @@ public HttpTransport(
       final int readTimeoutMillis,
       final boolean bypassSecurity,
       final @NotNull URL sentryUrl) {
+    this(
+        options,
+        connectionConfigurator,
+        connectionTimeoutMillis,
+        readTimeoutMillis,
+        bypassSecurity,
+        sentryUrl,
+        new CurrentDateProvider());
+  }
+
+  HttpTransport(
+      final @NotNull SentryOptions options,
+      final @NotNull IConnectionConfigurator connectionConfigurator,
+      final int connectionTimeoutMillis,
+      final int readTimeoutMillis,
+      final boolean bypassSecurity,
+      final @NotNull URL sentryUrl,
+      final @NotNull ICurrentDateProvider currentDateProvider) {
     this.proxy = options.getProxy();
     this.connectionConfigurator = connectionConfigurator;
     this.serializer = options.getSerializer();
     this.connectionTimeout = connectionTimeoutMillis;
     this.readTimeout = readTimeoutMillis;
     this.options = options;
     this.bypassSecurity = bypassSecurity;
+    this.currentDateProvider =
```

Comment:
m: Could be on a overload or inlined here where if the date isn't provided, you just new up inline. (or use a singleton since this will likely be used in a few more places).

The reasoning here is: We don't *need* this dependency. This is just for unit testing. In prod code we won't use anything other than the default one.

Reply:
mm not sure if I got it, the ctor that takes `currentDateProvider` is not public, the public one creates its own `currentDateProvider`, so its not a dependency, can you elaborate?

Reply:
I missed that, its resolved then. thanks